include/clang/AST/RecursiveASTVisitor.h

```diff
@@ -1263,6 +1268,7 @@ DEF_TRAVERSE_TYPELOC(FunctionProtoType, {
 DEF_TRAVERSE_TYPELOC(UnresolvedUsingType, {})
 DEF_TRAVERSE_TYPELOC(TypedefType, {})
 DEF_TRAVERSE_TYPELOC(TypeVariableType, {})
+DEF_TRAVERSE_TYPELOC(ExistentialType, {}) // TODO: is this correct?
```

Comment:
I believe this code needs to traverse the type locations for the type variable and the the body of the existing type, which are in the TypeLoc for the ExistentialType.

Reply:
Unfortunately, I still don't understand how type locations work. For example, this method (TraverseTypeLoc) sometimes calls itself recursively, but sometimes it calls TraverseType or even TraverseStmt. Maybe we can chat about this code tomorrow?